subworkflows/local/yaml_input/main.nf

```diff
@@ -26,30 +27,29 @@ workflow YAML_INPUT {
                 file(data.reference_file, checkIfExists: true),
             )
             map_order: data.map_order
-            read_ch: (data.assem_reads.read_type in ['hifi', 'clr', 'ont', 'illumina']
-                ? tuple(
-                    [
-                        id: tolid_ver,
-                        single_end: data.assem_reads.read_type != "illumina",
-                        read_type: data.assem_reads.read_type,
-                    ],
-                    data.assem_reads.read_data.collect { fq -> file(fq, checkIfExists: true, type: 'dir') },
-                )
-                : [])
+            read_ch: fn_get_validated_channel(
+                            "longread",
+                            tolid_ver,
+                            data.assem_reads.read_type,
+                            data.assembly.defined_class,
+                            data.assembly.project_id,
+                            data.assem_reads.read_data.collect()
```

Comment:
```suggestion
                            data.assem_reads.read_data
```
You shouldn't need `collect` here. It's just confusing to have a collection method that does nothing, especially one that has the same name as a channel operator.